src/transformers/models/opt/modeling_opt.py

```diff
@@ -359,9 +362,106 @@ def forward(
         return attn_output, attn_weights_reshaped, past_key_value
 
 
+class OPTSdpaAttention(OPTAttention):
+    """
+    OPT sdpa attention module. This module inherits from `OPTAttention` as the weights of the module stays untouched.
+    The only required change would be on the forward pass where it needs to correctly call the public API of sdpa
+    attention and deal with padding tokens in case the input contains any of them.
+    """
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        key_value_states: Optional[torch.Tensor] = None,
+        past_key_value: Optional[Tuple[torch.Tensor]] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        layer_head_mask: Optional[torch.Tensor] = None,
+        output_attentions: bool = False,
+        position_ids: Optional[torch.Tensor] = None,
+    ) -> Tuple[torch.Tensor, Optional[torch.Tensor], Optional[Tuple[torch.Tensor]]]:
+        if output_attentions or layer_head_mask is not None:
+            logger.warning_once("""OPTModel is using SDPA attention, which currently does not support output_attentions=True.
+                                   failing back to eager attention. remove warning using attn_implementation="eager".""")
+
```

Comment:
nit - formatting here is different from standard library patterns and elsewhere where the same warning is used 

```suggestion
            logger.warning_once(
                "OPTModel is using SDPA attention, which currently does not support output_attentions=True."
                'failing back to eager attention. remove warning using attn_implementation="eager".'
            )

```